tests/test_urls.py

```diff
@@ -119,15 +121,17 @@ def test_missing_parameters_for_new_issue(self):
 
     def test_new_issue_should_not_crash(self):
         '''Checks 500 is not accepted for /issues/new POST.'''
-        data = {'problem_category': u'mobile_site_bug',
-                'username': u'',
-                'description': u'foo',
-                'submit-type': u'github-proxy-report',
-                'url': u'http://example.com',
-                'os': u'Foobar',
-                'browser': u'BarFoo'}
-        rv = self.app.post('/issues/new', data=data)
-        self.assertNotEqual(rv.status_code, 500)
+        with patch('requests.post') as github_data:
+            github_data.status_code.return_value = 500
+            data = {'problem_category': u'mobile_site_bug',
+                    'username': u'',
+                    'description': u'foo',
+                    'submit-type': u'github-proxy-report',
+                    'url': u'http://example.com',
+                    'os': u'Foobar',
+                    'browser': u'BarFoo'}
+            rv = self.app.post('/issues/new', data=data)
+            self.assertNotEqual(rv.status_code, 500)
```

Comment:
@brizental ok got it!

in https://github.com/webcompat/webcompat.com/blob/586010a2e515c5ff451151f823684fcb99bc109f/webcompat/views.py#L175

If there is a form, 

```python
if request.form:
```

check a couple of things:

```python
        # To be legit the form needs a couple of parameters
        # if one essential is missing, it's a bad request
        must_parameters = set(['url', 'problem_category', 'description',
                               'os', 'browser',
                               'username', 'submit-type'])
        if not must_parameters.issubset(form.keys()):
            abort(400)
```

with our current test:

```python
        data = {'problem_category': u'mobile_site_bug',
                'username': u'',
                'description': u'foo',
                'submit-type': u'github-proxy-report',
                'url': u'http://example.com',
                'os': u'Foobar',
                'browser': u'BarFoo'}
```

we have all the parameters so it doesn't fail with a 400. This check was to handle gracefully  missing parameters with a 500. Let's remove a parameter.

```python
    def test_new_issue_should_not_crash(self):
        '''Checks 500 is not accepted for /issues/new POST.'''
        data = {'problem_category': u'mobile_site_bug',
                'username': u'',
                'submit-type': u'github-proxy-report',
                'url': u'http://example.com',
                'os': u'Foobar',
                'browser': u'BarFoo'}
        rv = self.app.post('/issues/new', data=data)
        self.assertEqual(rv.status_code, 400)
```

I removed `description`.
Let's check the status of issues in webcompat-tests

<img width="359" alt="capture d ecran 2017-08-11 a 09 46 16" src="https://user-images.githubusercontent.com/505230/29197853-0c57ecbe-7e7a-11e7-8972-386edd0f8221.png">

Latest issue is 1258.

```bash
nosetests --verbose --nocapture tests/test_urls.py
```
then the results:

```
Test that /about exists. ... ok
Test that asks user to log in before displaying activity. ... ok
Test POST to /csp-report w/ correct content-type returns 204. ... ok
Test POST w/ wrong content-type to /csp-report returns 400. ... ok
Test that the home page exists. ... ok
Test issues and integer for: ... ok
Test that the /issues/<number> exists, and does not redirect. ... ok
Test that the /issues route gets 200 and does not redirect. ... ok
Sends 400 to POST on /issues/new with missing parameters. ... here in request.form
form.keys(): ['url']
must test: False
ok
Test that /issues/new exists. ... ok
Checks 500 is not accepted for /issues/new POST. ... here in request.form
form.keys(): ['problem_category', 'username', 'submit-type', 'url', 'os', 'browser']
must test: False
ok
Test that /privacy exists. ... ok
Rate Limit URI sends 410 Gone. ... ok
Test that the /tools/cssfixme route gets 200. ... ok
Test that the /tools/cssfixme route gets 200 with ?url query. ... ok
Test that the /tools/cssfixme route gets 200 with bad ?url query. ... ok

----------------------------------------------------------------------
Ran 16 tests in 0.959s

OK
```

Everything passed. 🌈 Yeah. Let's check the `webcompat-tests`
No new issue created. Double 🌈🌈 yeah!  

Looking at the blame
https://github.com/webcompat/webcompat.com/blame/master/tests/test_urls.py#L120

My. fault. Sorry about that. Pfew.

@brizental let's fix it. That will be the last thing :)

```
'''/issues/new POST exit with 400 if missing parameters.'''
```